File: NiceReads/Backend/Server/tos-server/src/application/data/calculations/IScoreCalculator.ts
import { Uuid } from '../../../common/uuid';
import { SkillMatchingScore } from './SkillMatchingScore';

export interface IScoreCalculator {
    calculate(
        employeeSkills: Map<Uuid, Uuid>,
        roleSkills: Map<Uuid, Uuid>,
        skillLevels: Map<Uuid, number>): SkillMatchingScore;
}

File: EngagementHub/TOS.EngagementHub.Data/Calculations/IScoreCalculator.cs
using System;
using System.Collections.Generic;

namespace TOS.EngagementHub.Data.Calculations
{
    public interface IScoreCalculator
    {
        SkillMatchingScore Calculate(IReadOnlyDictionary<Guid, Guid> employeeSkills, IReadOnlyDictionary<Guid, Guid> roleSkills, IReadOnlyDictionary<Guid, int> skillLevels);
    }
}